docking_server:
  ros__parameters:

    #############################
    # General docking parameters
    #############################
    controller_frequency: 50.0
    initial_perception_timeout: 15.0
    wait_charge_timeout: 5.0
    dock_approach_timeout: 30.0

    undock_linear_tolerance: 0.05
    undock_angular_tolerance: 0.1
    
    max_retries: 3

    base_frame: base_link
    fixed_frame: odom  #
    dock_backwards: false
    dock_prestaging_tolerance: 0.5

    #############################
    # Dock plugins
    #############################
    dock_plugins: ['home_dock_plugin']

    home_dock_plugin:
      plugin: 'opennav_docking::SimpleChargingDock'
      docking_threshold: 0.05
      staging_x_offset: -0.7

      use_external_detection_pose: false # edited
      use_battery_status: false
      use_stall_detection: false

      external_detection_timeout: 0.0
      external_detection_translation_x: -0.18
      external_detection_translation_y: 0.0
      external_detection_rotation_roll: -1.57
      external_detection_rotation_pitch: -1.57
      external_detection_rotation_yaw: 0.0
      filter_coef: 0.1

    #############################
    # Dock database
    #############################
    docks: ['home_dock']

    home_dock:
      type: 'home_dock_plugin'
      frame: map
      pose: [2.28, 0.0, 0.25]

    #############################
    # Controller
    #############################
    controller:
      dock_collision_threshold: 0.5
      use_collision_detection: false  # look
      transform_tolerance: 0.5
      k_phi: 1.5
      k_delta: 1.0
      v_linear_min: 0.05
      v_linear_max: 0.15
